test/sim/abilities/magician.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe('Magician', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it.skip(`should not steal Weakness Policy on super-effective hits`, function () {
```

Comment:
It has to do with the order of checks done _after_ move success, which isn't well-documented yet. It's one of the things on my to-do list with the new USUM research. Suction Cups should be a check done _before_ move success though (if we're talking Suction Cups vs. Roar). 

Sort of like how accuracy works with prioritizing abilities > items, Weakness Policy should trigger using the same event, but after things like Flame Body / Rough Skin / Stamina. That event is sandwiched between things like Core Enforcer (before) and Jaboca Berry (after). Magician's timing is is alongside things like Beast Boost / Eject Button / Berserk, which is much later. Without the checks done _after_ move success properly sorted out yet though, this probably can be delayed.

Reply:
Actually I was probably thinking about Suction Cups vs. Dragon Tail on a Dragon-type with Weakness Policy, rather than against Roar, which gets processed differently.

Reply:
Aha, that makes more sense. Dragon Tail's switching effect is run after Weakness Policy, in the same event as things like Kee Berry or Berserk. So you would Dragon Tail, deal damage, trigger the Weakness Policy, and then be forced to switch out. But we're getting into the weeds with testing edge cases for checks done after move success; I'll add Dragon Tail vs. Weakness Policy but leave the general mechanic alone for this batch of tests.

Reply:
(The question I was actually asking was whether Suction Cups activates to prevent the forced switch before or after Weakness Policy.)